akka-sample-spring-extension-java/src/main/java/org/example/akka/di/SpringExtension.java

```diff
@@ -0,0 +1,33 @@
+package org.example.akka.di;
+
+import akka.actor.AbstractExtensionId;
+import akka.actor.ExtendedActorSystem;
+import akka.actor.Extension;
+import akka.actor.Props;
+import org.springframework.context.ApplicationContext;
+
+public class SpringExtension extends AbstractExtensionId<SpringExtension.SpringExt> {
+
+  public static SpringExtension SpringExtProvider = new SpringExtension();
+
+  @Override
+  public SpringExt createExtension(ExtendedActorSystem system) {
+    return new SpringExt();
+  }
+
+  public static class SpringExt implements Extension {
+    private volatile ApplicationContext applicationContext;
```

Comment:
This is the closest I found to the documentation. The only difference is that the Extension class is inside the AbstractExtensionId class.